# U.S. Medical Insurance Costs

Are there any trends associated with regionalisms? BMI? Children? Smoking?

Are there any non-regional trends that we see associated with regions? Sex? Age?

Are there trends that exist regardless of regionalisms?

In this step I imported the datafile. While we can look at the file in its entirety I think it is most useful to see here the headers.

In [6]:
import csv

with open('insurance.csv') as ins_master:
    ins_print = ins_master.readline()
    ins_line = ins_master.read()

print(ins_print)

age,sex,bmi,children,smoker,region,charges



Below are variables saved in their respective categories. Because we do not have a name to which to tie the individual pieces of data I have created a "person_ID" variable so that they information can be linked together.

In [7]:
with open('insurance.csv') as ins_csv:
  ins_reader = csv.DictReader(ins_csv, delimiter=',')
  age_list = []
  for row in ins_reader:
    age_list.append(row['age'])

In [8]:
with open('insurance.csv') as ins_csv:
  ins_reader = csv.DictReader(ins_csv, delimiter=',')
  sex_list = []
  for row in ins_reader:
    sex_list.append(row['sex'])

In [9]:
with open('insurance.csv') as ins_csv:
  ins_reader = csv.DictReader(ins_csv, delimiter=',')
  bmi_list = []
  for row in ins_reader:
    bmi_list.append(row['bmi'])

In [10]:
with open('insurance.csv') as ins_csv:
  ins_reader = csv.DictReader(ins_csv, delimiter=',')
  child_list = []
  for row in ins_reader:
    child_list.append(row['children'])

In [11]:
with open('insurance.csv') as ins_csv:
  ins_reader = csv.DictReader(ins_csv, delimiter=',')
  smoker_list = []
  for row in ins_reader:
    smoker_list.append(row['smoker'])

In [12]:
with open('insurance.csv') as ins_csv:
  ins_reader = csv.DictReader(ins_csv, delimiter=',')
  region_list = []
  for row in ins_reader:
    region_list.append(row['region'])

In [13]:
with open('insurance.csv') as ins_csv:
  ins_reader = csv.DictReader(ins_csv, delimiter=',')
  charges_list = []
  for row in ins_reader:
    charges_list.append(row['charges'])

In [14]:
person_ID = list(range(0, len(charges_list)))

In [15]:
def dictionary(ID, Age, Sex, BMI, Children, Smoker, Region, Charges):
  insurance_data = {}
  for index in range(len(person_ID)):
    insurance_data[person_ID[index]] = {"Person ID": person_ID[index], "Age": age_list[index], "Sex": sex_list[index], "BMI": bmi_list[index], "Children": child_list[index], "Smoker": smoker_list[index], "Region": region_list[index], "Charges": charges_list[index]}
  return insurance_data

In [16]:
ID_ins_dict = dictionary(0, 19 , 'female', 27.9 , 0 , 'yes', 'southwest', '16884.924')

In [17]:
print(ID_ins_dict)

{0: {'Person ID': 0, 'Age': '19', 'Sex': 'female', 'BMI': '27.9', 'Children': '0', 'Smoker': 'yes', 'Region': 'southwest', 'Charges': '16884.924'}, 1: {'Person ID': 1, 'Age': '18', 'Sex': 'male', 'BMI': '33.77', 'Children': '1', 'Smoker': 'no', 'Region': 'southeast', 'Charges': '1725.5523'}, 2: {'Person ID': 2, 'Age': '28', 'Sex': 'male', 'BMI': '33', 'Children': '3', 'Smoker': 'no', 'Region': 'southeast', 'Charges': '4449.462'}, 3: {'Person ID': 3, 'Age': '33', 'Sex': 'male', 'BMI': '22.705', 'Children': '0', 'Smoker': 'no', 'Region': 'northwest', 'Charges': '21984.47061'}, 4: {'Person ID': 4, 'Age': '32', 'Sex': 'male', 'BMI': '28.88', 'Children': '0', 'Smoker': 'no', 'Region': 'northwest', 'Charges': '3866.8552'}, 5: {'Person ID': 5, 'Age': '31', 'Sex': 'female', 'BMI': '25.74', 'Children': '0', 'Smoker': 'no', 'Region': 'southeast', 'Charges': '3756.6216'}, 6: {'Person ID': 6, 'Age': '46', 'Sex': 'female', 'BMI': '33.44', 'Children': '1', 'Smoker': 'no', 'Region': 'southeast', 'Cha

ID_ins_dict is the full dictionary of all of the results of the csv provided. I have set a key with an assigned ID number (0-1337) to tabulate the information. Now I should be able to run this information through functions or classes.

In [106]:
ID_ins_dict[0]

{'Person ID': 0,
 'Age': '19',
 'Sex': 'female',
 'BMI': '27.9',
 'Children': '0',
 'Smoker': 'yes',
 'Region': 'southwest',
 'Charges': '16884.924'}

In [108]:
region_list = [ID_ins_dict.get(person,{}).get('Region') for person in ID_ins_dict.keys()]

In [109]:
BMI_list = [ID_ins_dict.get(person,{}).get('BMI') for person in ID_ins_dict.keys()]

In [110]:
sex_list = [ID_ins_dict.get(person,{}).get('Sex') for person in ID_ins_dict.keys()]

In [111]:
age_list = [ID_ins_dict.get(person,{}).get('Age') for person in ID_ins_dict.keys()]

In [112]:
smoker_list = [ID_ins_dict.get(person,{}).get('Smoker') for person in ID_ins_dict.keys()]

In [113]:
child_list = [ID_ins_dict.get(person,{}).get('Children') for person in ID_ins_dict.keys()]

In [107]:
charges_list = [ID_ins_dict.get(person,{}).get('Charges') for person in ID_ins_dict.keys()]

In [156]:
region_BMI = list(zip(region_list, BMI_list))

In [121]:
region_child = list(zip(region_list, child_list))

In [133]:
region_charges = list(zip(region_list, charges_list))

In [139]:
region_age = list(zip(region_list, age_list))

In [146]:
region_sex = list(zip(region_list, sex_list))

In [157]:
region_smoker = list(zip(region_list, smoker_list))

In [147]:
def region_specifics(big_list, list_type):
    southwest = 0
    southwest_count = 0
    northwest = 0
    northwest_count = 0
    southeast = 0
    southeast_count = 0
    northeast = 0
    northeast_count = 0
    for item in big_list:
        if item[0] == 'southwest':
            southwest += float(item[1])
            southwest_count += 1
        elif item[0] == 'northwest':
            northwest += float(item[1])
            northwest_count += 1
        elif item[0] == 'southeast':
            southeast += float(item[1])
            southeast_count += 1
        elif item[0] == 'northeast':
            northeast += float(item[1])
            northeast_count += 1
    return 'Southwest ' + list_type + ' average is: ' + str(southwest/southwest_count) + '. Northwest ' + list_type + ' average is: ' + str(northwest/northwest_count) + '. Southeast ' + list_type + ' average is: ' + str(southeast/southeast_count) + '. Northeast ' + list_type + ' average is: ' + str(northeast/northeast_count) + '.' 

In [154]:
def region_specifics_word(big_list):
    southwest_male = 0
    southwest_female = 0
    southwest_count = 0
    northwest_male = 0
    northwest_female = 0
    northwest_count = 0
    southeast_male = 0
    southeast_female = 0
    southeast_count = 0
    northeast_male = 0
    northeast_female = 0
    northeast_count = 0
    for item in big_list:
        if item[0] == 'southwest':
            southwest_count += 1
            if item[1] == 'male':
                southwest_male += 1
            else:
                southwest_female += 1            
        elif item[0] == 'northwest':
            northwest_count += 1
            if item[1] == 'male':
                northwest_male += 1
            else:
                northwest_female += 1 
        elif item[0] == 'southeast':
            southeast_count += 1
            if item[1] == 'male':
                southeast_male += 1
            else:
                southeast_female += 1
        elif item[0] == 'northeast':
            northeast_count += 1
            if item[1] == 'male':
                northeast_male += 1
            else:
                northeast_female += 1
    return 'Southwest gender breakdown is ' + str((southwest_male/southwest_count)*100) + '% male and ' + str((southwest_female/southwest_count)*100) + '% female. Northwest gender breakdown is ' + str((northwest_male/northwest_count)*100) + '% male and ' + str((northwest_female/northwest_count)*100) + '% female. Southeast gender breakdown is ' + str((southeast_male/southeast_count)*100) + '% male and ' + str((southeast_female/southeast_count)*100) + '% female. Northeast gender breakdown is ' + str((northeast_male/northeast_count)*100) + '% male and ' + str((northeast_female/northeast_count)*100) + '% female.'

In [158]:
def region_specifics_word(big_list):
    southwest_smoker = 0
    southwest_nonsmoker = 0
    southwest_count = 0
    northwest_smoker = 0
    northwest_nonsmoker = 0
    northwest_count = 0
    southeast_smoker = 0
    southeast_nonsmoker = 0
    southeast_count = 0
    northeast_smoker = 0
    northeast_nonsmoker = 0
    northeast_count = 0
    for item in big_list:
        if item[0] == 'southwest':
            southwest_count += 1
            if item[1] == 'yes':
                southwest_smoker += 1
            else:
                southwest_nonsmoker += 1            
        elif item[0] == 'northwest':
            northwest_count += 1
            if item[1] == 'yes':
                northwest_smoker += 1
            else:
                northwest_nonsmoker += 1 
        elif item[0] == 'southeast':
            southeast_count += 1
            if item[1] == 'yes':
                southeast_smoker += 1
            else:
                southeast_nonsmoker += 1
        elif item[0] == 'northeast':
            northeast_count += 1
            if item[1] == 'yes':
                northeast_smoker += 1
            else:
                northeast_nonsmoker += 1
    return 'Southwest smoker breakdown is ' + str((southwest_smoker/southwest_count)*100) + '% smoker and ' + str((southwest_nonsmoker/southwest_count)*100) + '% non-smoker. Northwest smoker breakdown is ' + str((northwest_smoker/northwest_count)*100) + '% smoker and ' + str((northwest_nonsmoker/northwest_count)*100) + '% non-smoker. Southeast smoker breakdown is ' + str((southeast_smoker/southeast_count)*100) + '% smoker and ' + str((southeast_nonsmoker/southeast_count)*100) + '% non-smoker. Northeast smoker breakdown is ' + str((northeast_smoker/northeast_count)*100) + '% smoker and ' + str((northeast_nonsmoker/northeast_count)*100) + '% non-smoker.'

In [143]:
print(region_specifics(region_charges, 'charges'))

Southwest charges average is: 12346.93737729231. Northwest charges average is: 12417.575373969228. Southeast charges average is: 14735.411437609895. Northeast charges average is: 13406.3845163858.


This shows us that the Southeast (14,735) has the highest average charges and the Southwest (12,347) has the lowest average charges.
Let's take a look at some of the regional trends.

In [141]:
print(region_specifics(region_BMI, 'BMI'))

Southwest BMI average is: 30.59661538461538. Northwest BMI average is: 29.199784615384626. Southeast BMI average is: 33.35598901098903. Northeast BMI average is: 29.17350308641976.


In [142]:
print(region_specifics(region_child, 'number of children'))

Southwest number of children average is: 1.1415384615384616. Northwest number of children average is: 1.1476923076923078. Southeast number of children average is: 1.0494505494505495. Northeast number of children average is: 1.0462962962962963.


In [159]:
region_specifics_word(region_smoker)

'Southwest smoker breakdown is 17.846153846153847% smoker and 82.15384615384616% non-smoker. Northwest smoker breakdown is 17.846153846153847% smoker and 82.15384615384616% non-smoker. Southeast smoker breakdown is 25.0% smoker and 75.0% non-smoker. Northeast smoker breakdown is 20.679012345679013% smoker and 79.32098765432099% non-smoker.'

The Southwest (30.6) and Southeast (33.4) have the highest average BMI. The Northeast and Northwest both have the same (29.2) BMI.

The Southeast (25%) and Northeast (20.7%) have the highest percentage of smokers. The Southwest and Northwest both have the same (17.8%) of smokers.

Each region averages roughly the same amount of children (1) per person. So we can conclude the number of children is not a deciding factor within regional trends.

In [144]:
print(region_specifics(region_age, 'age'))

Southwest age average is: 39.45538461538462. Northwest age average is: 39.19692307692308. Southeast age average is: 38.93956043956044. Northeast age average is: 39.26851851851852.


In [155]:
region_specifics_word(region_sex)

'Southwest gender breakdown is 50.153846153846146% male and 49.84615384615385% female. Northwest gender breakdown is 49.53846153846154% male and 50.46153846153846% female. Southeast gender breakdown is 51.92307692307693% male and 48.07692307692308% female. Northeast gender breakdown is 50.308641975308646% male and 49.691358024691354% female.'

The ages average roughly the same in each region - 39 years old.

The gender breakdown varries slightly, but hovers around a 50/50 male to female split.